src/librustc_metadata/schema.rs

```diff
@@ -199,15 +199,9 @@ pub struct CrateRoot<'tcx> {
 
 }
 
-/// Stores extra information about proc macros
-/// that would be difficult to store along with
-/// the dylib symbol.
-/// Currently, this is only used by rustdoc,
-/// which needs access to the proc macro span and
-/// attributes for inlining purposes
 #[derive(RustcEncodable, RustcDecodable)]
 pub struct ProcMacroData {
-    pub item: DefId,
+    pub item: DefIndex,
```

Comment:
I don't think you need the struct anymore btw, you could just have `proc_macros: Option<LazySeq<DefIndex>>`.

And then use something like this to check whether something is a proc macro:
```rust
self.root.proc_macros.map_or(false, |xs| xs.decode(self).find(|x| x == def_id.index))
```